### PR TITLE
Prevent short lived connections appearing as pseudo nodes in the applications view.

### LINKED ISSUE #447: Short lived connections are showing up as pseudo nodes in the the applications view.
screen shot 2015-09-07 at 11 07 03

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #447

### CODE REVIEW

**Comment by peterbourgon** [comment]
For context,

```
<Tom> For short lived connections, we don't know the pid, so they can't be merged
<Tom> so they just shouldn't be there
<Tom> which is what the PR does; filters out shortlived connections which have no pids
<Peter> Ah, and in the containers view, it's OK, because while we don't know the PID we do know the [IP address, which is enough to join to a container]
```

File: render/mapping.go [hunk 1]
```diff
@@ -45,9 +45,16 @@ func MapEndpointIdentity(m RenderableNode, local report.Networks) RenderableNode
 		return RenderableNodes{}
 	}
 
+	// We need to filter out short lived connections from this view,
+	// if they don't also have a pid; see #447
+	pid, pidOK := m.Metadata[process.PID]
+	_, conntracked := m.Metadata[endpoint.Conntracked]
+	if !pidOK && conntracked {
```
Comment (peterbourgon) [inline render/mapping.go]: OK, and you only squelch the node if you see it's conntracked, otherwise pseudo is created.

**Comment by peterbourgon** [comment]
LGTM.

File: render/mapping.go (hunk 1 above)
Comment (tomwilkie) [inline render/mapping.go]: Yes the previous behaviour (from procspy) is preserved - pseudo nodes are created for non-contracked connections.

**Pull request merged**